File: backend/Wms.Api/Domain/Entities/Location.cs
using Wms.Api.Domain.Enums;

namespace Wms.Api.Domain.Entities;

public class Location
{
    public int Id { get; set; }
    public int WarehouseId { get; set; }
    public required string Code { get; set; }
    public required string Name { get; set; }
    public LocationType Type { get; set; } = LocationType.Storage;
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public Warehouse Warehouse { get; set; } = null!;
}

File: frontend/wms-web/src/types/location.ts
export type LocationType = 'Receiving' | 'Storage' | 'Shipping'

export type Location = {
  id: number
  warehouseId: number
  warehouseName: string
  code: string
  name: string | null
  type: LocationType
  isActive: boolean
  createdAtUtc: string
}

export type CreateLocationRequest = {
  code: string
  name: string | null
  type: LocationType
}

export type UpdateLocationRequest = CreateLocationRequest & {
  isActive: boolean
}
